api/types/registry/registry.go

```diff
@@ -102,3 +105,14 @@ type SearchResults struct {
 	// Results is a slice containing the actual results for the search
 	Results []SearchResult `json:"results"`
 }
+
+// ManifestInspect describes the result obtained from contacting the
+// registry to retrieve image metadata
+// TODO(nishanttotla): Decide which other fields to add here
+type ManifestInspect struct {
+	// Digest is the content addressable digest for the image on the registry
+	Digest digest.Digest
```

Comment:
The manifest is a JSON document. `DeserializedManifest` is just a convenience for working with it in Go code. If you marshal it to JSON, you'll get the same set of bytes back, so the hash won't change (`DeserializedManifest` implements a `MarshalJSON` method that ensures this).

However, I've thought about it some more, and I don't think using `[]byte` is worth the ugliness. Let's do this instead:

```go
{
  Digest       digest.Digest,
  Manifest     *json.RawMessage,
  ConfigJSON   *json.RawMessage,
}
```

`Manifest` would contain the original manifest JSON returned by `(*DeserializedManifest).MarshalJSON`. This is similar to using `[]byte`, but it causes the JSON to be inserted directly instead of as a string. We lose the ability to verify the digest (because of possible whitespace added around the value during the marshalling process), but we avoid the ugliness off putting JSON inside a string inside JSON.

> Why does the digest need to be verified at the client?

I don't think there's an immediate reason to verify the digest at the client, but I was thinking that if we're going to send the manifest to the client, we might as well send the exact manifest.

> Why does sending []bytes help with it?

Because the hash is computed over these bytes.

> > It's ugly to put JSON inside JSON though

> This applies to ConfigJSON as well, right?

Yes.